packages/lodestar/src/network/gossip/validation/aggregateAndProof.ts

```diff
@@ -27,45 +28,44 @@ export async function validateGossipAggregateAndProof(
   const logContext = getLogContext(aggregate, aggregateAndProof, root, attestationRoot);
   logger.verbose("Started gossip aggregate and proof validation", logContext);
   if (!hasValidAttestationSlot(config, chain.clock.currentSlot, aggregate.data.slot)) {
-    logger.warn("Ignored gossip aggregate and proof", {
-      reason: "invalid slot time",
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_SLOT_OUT_OF_RANGE,
       currentSlot: chain.clock.currentSlot,
       ...logContext,
+      job: attestationJob,
     });
     // TODO: aggregate and proof pool to wait until proper slot to replay
-    return ExtendedValidatorResult.ignore;
   }
   if (await db.seenAttestationCache.hasAggregateAndProof(aggregateAndProof)) {
-    logger.warn("Ignored gossip aggregate and proof", {
-      reason: "seen aggregator and target index",
-      targetEpoch: aggregate.data.target.epoch,
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_AGGREGATE_ALREADY_KNOWN,
       ...logContext,
+      root: attestationRoot,
+      job: attestationJob,
     });
-    return ExtendedValidatorResult.ignore;
   }
   if (!hasAttestationParticipants(aggregate)) {
-    logger.warn("Rejected gossip aggregate and proof", {
-      reason: "missing attestation participants",
+    // missing attestation participants
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_WRONG_NUMBER_OF_AGGREGATION_BITS,
       ...logContext,
+      job: attestationJob,
     });
-    return ExtendedValidatorResult.reject;
   }
-
   if (await isAttestingToInValidBlock(db, aggregate)) {
-    logger.warn("Rejected gossip aggregate and proof", {
-      reason: "attesting to invalid block",
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_KNOWN_BAD_BLOCK,
       ...logContext,
```

Comment:
but logContext type isn't included in AttestationErrorType. I think you should include it.
Not sure how you ain't getting type errors.